README.md

```diff
@@ -163,594 +104,281 @@ The invoker MUST be the UCAN delegator. Their DID MUST be authenticated in the `
-      "subject": "Coffee"
-    }
-  },
-  "bafy...multipleAuth": {
-    "scope": [
-      {
-        "/": "bafy...sendEmailTask"
-      },
-      {
-        "/": "bafy...createBlogPostTask"
-      }
-    ],
-    "s": {
-      "/": {
-        "bytes": "7aEDQMyGqYw2iwP7uIn+Kav5AWe9l5VnL72Gpkzs1Azp+zs6vnixQPa1aCSrok4XwKkhSlFRmRN8YbyohB6iDFl4CQ8"
-      }
+  "sig": {"/": {bytes: "7aEDQIscUKVuAIB2Yj6jdX5ru9OcnQLxLutvHPjeMD3pbtHIoErFpo7OoC79Oe2ShgQMLbo2e6dvHh9scqHKEOmieA0"}},
+  "inv": cid({
+    "iss": "did:plc:ewvi7nxzyoun6zhxrhs64oiz",
+    "aud": "did:key:z6MkrZ1r5XBFZjBU34qyD8fueMbMRkKw17BZaq2ivKFjnz2z",
+    "run": cid({
+      "act": {
+        "nnc": "email-akiko#1234567890"
+        "cmd": "msg/send",
+        "arg": {
+          "from": "mailto:akiko@example.com",
+          "to": [ "boris@example.com", "carol@example.com" ],
+          "subject": "Coffee",
+          "body": "Let get coffee sometime and talk about UCAN Invocations!"
+        }
+      }),
+      "mta": {},
+      "prf": [{"/": "bafkr4iblvgvkmqt46imsmwqkjs7p6wmpswak2p5hlpagl2htiox272xyy4"}],
+      "exp": 1697409438
     }
-  },
-  "bafy...createBlogPostInvocation": {
-    "v": "0.1.0",
-    "run": {
-      "/": "bafy...createBlogPostTask"
-    },
-    "auth": {
-      "/": "bafy...multipleAuth"
-    },
-    "prf": [
-      {
-        "/": "bafyreid6q7uslc33xqvodeysekliwzs26u5wglas3u4ndlzkelolbt5z3a"
-      }
-    ]
-  },
-  "bafy...sendEmailInvocation": {
-    "v": "0.1.0",
-    "run": {
-      "/": "bafy...sendEmailTask"
-    },
-    "auth": {
-      "/": "bafy...multipleAuth"
-    },
-    "prf": [
-      {
-        "/": "bafyreihvee5irbkfxspsim5s2zk2onb7hictmpbf5lne2nvq6xanmbm6e4"
-      }
-    ]
-  }
+  })
 }
 ```
 
-### 5.3.3 Causal Invocations
+### 3.4.3 Inline WebAssembly
 
-```json
+```js
 {
-  "bafy...updateDnsTask": {
-    "on": "dns:example.com?TYPE=TXT",
-    "call": "crud/update",
-    "input": {
-      "value": "hello world"
-    }
-  },
-  "bafy...auth": {
-    "scope": [
-      {
-        "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
-      }
-    ],
-    "s": {
-      "/": {
-        "bytes": "7aEDQIscUKVuAIB2Yj6jdX5ru9OcnQLxLutvHPjeMD3pbtHIoErFpo7OoC79Oe2ShgQMLbo2e6dvHh9scqHKEOmieA0"
-      }
-    }
-  },
-  "bafy...updateDnsInvocation": {
-    "v": "0.1.0",
-    "run": {
-      "/": "bafy...updateDnsTask"
-    },
-    "auth": {
-      "/": "bafy...auth"
-    },
-    "cause": {
-      "/": "bafy...somePriorInvocation"
-    },
-    "prf": [
-      {
-        "/": "bafyreieynwqrabzdhgl652ftsk4mlphcj3bxchkj2aw5eb6dc2wxieilau"
+  "sig": {"/": {bytes: "7aEDQIscUKVuAIB2Yj6jdX5ru9OcnQLxLutvHPjeMD3pbtHIoErFpo7OoC79Oe2ShgQMLbo2e6dvHh9scqHKEOmieA0"}},
+  "inv": cid({
+    "iss": "did:plc:ewvi7nxzyoun6zhxrhs64oiz",
+    "aud": "did:key:z6MkrZ1r5XBFZjBU34qyD8fueMbMRkKw17BZaq2ivKFjnz2z",
+    "run": ({
+      "act": {
+        "nnc": "", // NOTE: as stated above, idempotent Actions should always have the same nonce
+        "act": "wasm/run",
```

Comment:
act within act is weird IMO. Shouldn't this be the `cmd` and that is something we can match against with any other invocation? It feels like something that should be standard so we don't have to be like look inside act, is it a cmd, or another act, etc.